panel/applets/sensors/SensorsApplet.vala

```diff
@@ -0,0 +1,210 @@
+  {
+          Object(uuid: uuid);
+
+          settings_schema = "com.solus-project.sensors";
+          settings_prefix = "/com/solus-project/budgie-panel/instance/sensors";
+          m_settings = this.get_applet_settings(uuid);
+          m_settings.changed.connect(on_settings_change);
+
+          m_widget = new Gtk.EventBox();
+          m_label = new Gtk.Label(m_resultString);
+          m_widget.add(m_label);
+          m_label.show();
+
+          create_menuMain();
+          m_popover = new Gtk.Popover.from_model(m_widget, m_menuMain);
+          m_popover.get_child().show_all();
+
+          m_widget.button_press_event.connect((e)=> {
+                          if (e.button != 3) {
+                                  return Gdk.EVENT_PROPAGATE;
+                          }
+                          if (m_popover.get_visible()) {
+                                  m_popover.hide();
+                          } else {
+                                  m_manager.show_popover(m_widget);
+                          }
+                          return Gdk.EVENT_STOP;
+                  });
+
+          getSensors();
+          add(m_widget);
+          show_all();
+
+          m_running = true;
+          Timeout.add_seconds_full(GLib.Priority.LOW, 1, getSensors);
+  }
+
+  public override void update_popovers(Budgie.PopoverManager ? manager)
+  {
+          m_manager = manager;
+          m_manager.register_popover(m_widget, m_popover);
+  }
+
+  void on_settings_change(string key)
+  {
+          if ("unit" == key)
+          {
+                  int unit = m_settings.get_int(key);
+                  switch (unit)
+                  {
+                  case 0 :
+                          m_isCelsius = true;
+                          break;
+
+                  case 1 :
+                          m_isCelsius = false;
+                          break;
+                  }
+          }
+  }
+
+  public void create_menuMain()
+  {
+          m_menuMain = new GLib.Menu();
+          m_menuMain.append ("Celsius", "celsius");
+          m_menuMain.append ("Fahrenheit", "fahrenheit");
+
+          var celsius_action = new SimpleAction ("celsius", null);
+          celsius_action.activate.connect (this.celsius_cb);
+
+          var fahrenheit_action = new SimpleAction ("fahrenheit", null);
+          fahrenheit_action.activate.connect (this.fahrenheit_cb);
+  }
+
+
+  void celsius_cb (SimpleAction simple, Variant ? parameter) {
+          m_isCelsius = true;
+          m_settings.set_int ("unit", 0);
+  }
+
+  void fahrenheit_cb (SimpleAction simple, Variant ? parameter) {
+          m_isCelsius = false;
+          m_settings.set_int ("unit", 1);
+  }
+
+  private bool getSensors()
+  {
+          string ls_stdout;
+          string ls_stderr;
+          int ls_status;
+          string editStr = "";
+
+          string [] lines;
+          string [] elines;
+          string [] e2lines;
+
+          lock (m_lock)
```

Comment:
Lock and `spawn_command_line_sync` ? You're blocking the whole panel thread and breaking the desktop here. Use concurrency if necessary